### PR TITLE
ENH: add docstring of butter() for transfer function syntax problem

### LINKED ISSUE #9197: problem with scipy.signal.butter with 1000+ points array
### Please describe the issue in detail here, and for bug reports fill in the fields below.

When I produce a filter from the butter function, it works all fine for up to 1000 points.
Then trouble starts to happen somehow... The problem shows very clearly starting at 8000 points onwards, 10000 has it real bad and it only get worst
the filter has higher transfer function values than 1, and furthermore has a rather chaotic behaviour. The signal is therefore not filtered properly.

### Reproducing code example:

<<A short code example that reproduces the problem. It should be self-contained, i.e., possible to run as-is via 'python myproblem.py'>>

Here's a simple enough function and code to play with the number of points in the input

```
import numpy as np
from numpy import sin
from scipy.signal import butter, freqz, lfilter
import matplotlib.pyplot as plt



def show_the_problem(point_number):
    #we will show the problem here
    # our "data"
    x = np.linspace(0, 2*np.pi, point_number)
    y = sin(x) + 0.3* sin(10*x)
    
    # sampling frequency would be 1/ sampling_time
    sampling_frequency = point_number/(2*np.pi)
    # hence the nyquist frequency
    nyq = sampling_frequency/2
    # desired cutoff frequency, in Hertz
    cutoff = 1
    # normalized for the function butter
    normalized_cutoff = cutoff/nyq

    b,a = butter(6, normalized_cutoff)
    w, h = freqz(b, a, point_number)
    #filtered data
    y_filtered = lfilter(b, a, y)
    
    
     
    # plotting part
    plt.close("all")
    plt.figure()
    thefigure = plt.gcf()
    thefigure.subplots_adjust(hspace=1)
    plt.subplot(3,1,1)
    plt.plot(x, y)
    plt.title("'data'")
    plt.subplot(3,1,2)
    plt.xlim(0, cutoff * 10)
    plt.plot(nyq * w /np.pi, np.abs(h))
    plt.title("filter response")
    plt.subplot(3,1,3)
    plt.plot(x, y_filtered)
    plt.title("filtered data")
    
    return None
```

### Error message:
as a matter of fact, if i try with a very large number of points, like 100000, then i get this error:
```
<<Full error message, if any (starting from line Traceback: ...)>>

show_the_problem(100000)
/opt/anaconda3/lib/python3.6/site-packages/scipy/signal/filter_design.py:444: RuntimeWarning: divide by zero encountered in true_divide
  npp_polyval(zm1, a, tensor=False))
/opt/anaconda3/lib/python3.6/site-packages/scipy/signal/filter_design.py:444: RuntimeWarning: invalid value encountered in true_divide
  npp_polyval(zm1, a, tensor=False))
```

### Scipy/Numpy/Python version information:
```
<<Output from 'import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info)'>>
1.0.0 1.14.0 sys.version_info(major=3, minor=6, micro=4, releaselevel='final', serial=0)
```

### PR DESCRIPTION
**Pull request by AtsushiSakai**
#### Reference issue
To fix #9197

#### What does this implement/fix?
add docstring of butter() for transfer function syntax problem

### CODE REVIEW

File: scipy/signal/filter_design.py [hunk 1]
```diff
@@ -2842,6 +2842,10 @@ def butter(N, Wn, btype='low', analog=False, output='ba', fs=None):
 
     The ``'sos'`` output parameter was added in 0.16.0.
 
+    If you use the transfer function syntax [b, a], you might encounter numerical problems.
```
Comment (ilayn) [inline scipy/signal/filter_design.py]: I would suggest a bit more informative version

```
    The conversion between roots and polynomial coefficients is a numerically 
    sensitive operation. Thus, conversion to a Transfer Function can be unstable
    even for N = 4 and above. It is recommended to keep the SOS representation.
```

or something along those lines. 

Also we need to keep an eye on the PEP8 stuff. The line is longer than 80 characters.

**Review by ilayn — COMMENTED** [review]
Nice! Thank you for the effort @AtsushiSakai keep them coming. A minor comment.

File: scipy/signal/filter_design.py (hunk 1 above)
Comment (AtsushiSakai) [reply scipy/signal/filter_design.py]: @ilayn Thank you for your review. I fixed it based on your review. Please check it again.

**Comment by ilayn** [comment]
I would suggest the following for less verbosity and it is better if we skip the matlab reference. Not only it can disappear but also it is probably the 2nd place people would look anyways. 

```
    If the transfer function form ``[b, a]`` is requested, numerical
    problems can occur since the conversion between roots and
    the polynomial coefficients is a numerically sensitive operation,
    even for N >= 4. It is recommended to work with the SOS
    representation.
```

Just for the future reference, please make separate branches to work on new features and to send PRs such that your standard working repo does not interfere with the PRs you have submitted. Once a PR is merged you can safely delete that branch and keep working on other branches.

**Comment by AtsushiSakai** [comment]
@ilayn OK. I fixed the docstring.

**Comment by AtsushiSakai** [comment]
>Just for the future reference ..
Thank you so much. I will do that from the next PR.

**Comment by ilayn** [comment]
Thanks @AtsushiSakai !

**Pull request merged**